### ISSUE TITLE
`Unused constructor dependencies`: falsely faulted when actually used in trait

### ISSUE THREAD

**Issue by afoeder** [kalessil/phpinspectionsea#544]
I almost don't want to mention that because I would confess using traits; but given…

```
<?php
trait SomeTrait
{
    private fuction foo()
    {  $this->dependency->do(); }
}

class Adopter
{
    use SomeTrait;
    public function __construct(Dependency $dependency)
    {
        $this->d̲e̲p̲e̲n̲d̲e̲n̲c̲y̲ = $dependency; // "Unused ctor deps" inspection kicks in
    }
}
```

![image](https://user-images.githubusercontent.com/1125168/30633417-da1a359a-9deb-11e7-9481-7c75c0490bfa.png)

I am really ashamed that this occurs in bad code style environments only but still a valid issue IMO.

Thank you for your awesome work @kalessil !

**Comment by kalessil** [comment]
Aside of the evil nature of traits, this is a bug, so will be fixed in the next release (2.3.11 xD) =)

**Comment by kalessil** [comment]
Labeling as question: cannot reproduce (added a new test-case).
@afoeder: something else is triggering this - what I find strange is that the property doesn't belong to the trait, perhaps relocating the property into the trait (dropping in related classes) will help.
